src/c2s/mongoose_c2s.erl

```diff
@@ -413,14 +424,38 @@ handle_sasl_step(StateData, {error, NewSaslAcc, Result}, Retries) ->
     handle_sasl_error(StateData, NewSaslAcc, Result, Retries).
 
 -spec handle_sasl_success(data(), mongoose_acc:t(), mongoose_c2s_sasl:success()) -> fsm_res().
-handle_sasl_success(StateData = #c2s_data{info = Info}, SaslAcc,
-                    #{server_out := MaybeServerOut, jid := Jid, auth_module := AuthMod}) ->
-    StateData1 = StateData#c2s_data{streamid = new_stream_id(), jid = Jid,
-                                    info = maps:merge(Info, #{auth_module => AuthMod})},
-    El = mongoose_c2s_stanzas:sasl_success_stanza(MaybeServerOut),
-    send_acc_from_server_jid(StateData1, SaslAcc, El),
-    ?LOG_INFO(#{what => auth_success, text => <<"Accepted SASL authentication">>, c2s_state => StateData1}),
-    {next_state, {wait_for_stream, authenticated}, StateData1, state_timeout(StateData1)}.
+handle_sasl_success(StateData = #c2s_data{jid = MaybeInitialJid, info = Info}, SaslAcc,
+                    #{server_out := MaybeServerOut, jid := SaslJid, auth_module := AuthMod}) ->
+    case verify_initial_jid(MaybeInitialJid, SaslJid) of
+        true ->
+            StateData1 = StateData#c2s_data{streamid = new_stream_id(), jid = SaslJid,
+                                            info = maps:merge(Info, #{auth_module => AuthMod})},
+            El = mongoose_c2s_stanzas:sasl_success_stanza(MaybeServerOut),
+            send_acc_from_server_jid(StateData1, SaslAcc, El),
+            ?LOG_INFO(#{what => auth_success, text => <<"Accepted SASL authentication">>, c2s_state => StateData1}),
+            {next_state, {wait_for_stream, authenticated}, StateData1, state_timeout(StateData1)};
+        false ->
+            c2s_stream_error(StateData, mongoose_xmpp_errors:invalid_from())
+    end.
+
+% 6.4.6.  SASL Success
```

Comment:
I'm just wondering if it makes sense to copy-paste such chunks into the code. Maybe a link would be sufficient? I think it's harder to find the RFC version from this chunk than to find the content from a link. Of course a link can die, but hopefully RFC ones don't do it often.